### ISSUE TITLE
govendor fetch - multiple canonical deps

### ISSUE THREAD

**Issue by tejasmanohar** [kardianos/govendor#288]
I often get errors about having multiple canonical dependencies and then they go away after running the same `govendor fetch +external` command 2-4 times. Does govendor fetch do something that's not recoverable on error?

**Comment by nalbion** [comment]
I'm having similar issues.  I'm pretty sure that I'm doing something fundamentally wrong.

- I've created an application project and two go libraries
- one of the libraries depends on the other library and
- the application depends on both libraries
- all 3 projects use govendor and I commit the `vendor/` directory an all of its contents to source control.
  (so that I can run tests for the libraries in CI)

I'm a bit confused by how I'm supposed to use govendor.  When I make changes in my libraries the (somewhat random) process that I follow doesn't seem right:
- /go/src/github.com/nalbion/my-library $ git add * && git commit -m 'update' && git push
- /go/src/github.com/nalbion/my-app $ rm -rf vendor/github.com/nalbion/my-library
- /go/src/github.com/nalbion/my-app $ govendor update +external

...but then I end up with entries like this in vendor.json:

```
               {
			"origin": "github.com/nalbion/my-library/vendor/github.com/nalbion/my-other-library/foo/bar",
			"path": "github.com/nalbion/my-other-library/foo/bar",
			"revision": "...",
			"revisionTime": "2017-03-01T14:14:47Z"
		},
                {
			"origin": "github.com/nalbion/my-library/vendor/github.com/aws/aws-sdk-go/private/protocol/rest",
			"path": "github.com/aws/aws-sdk-go/private/protocol/rest",
			"revision": "769d3986e8366845e7073f320376c129cf865d39",
			"revisionTime": "2017-03-01T13:55:18Z"
		},
```

...why is govendor including the vendor folder of my library?

**Comment by nalbion** [comment]
some of my questions have been answered by the [FAQ](https://github.com/kardianos/govendor/blob/master/doc/faq.md) and [whitepaper](https://github.com/kardianos/govendor/blob/master/doc/whitepaper.md)

These 2 rules in the [Developer Guide](https://github.com/kardianos/govendor/blob/master/doc/dev-guide.md) seem contradictory - what happens if the dependencies of one of my libraries is removed from github?  I'd have to recover them from an application which had vendored my library (and its dependencies)
> - Do not check-in vendor sources if you expect an external package to import it
> - Do check-in vendor sources for main packages

**Comment by tejasmanohar** [comment]
@nalbion I agree. If you're sharing code betweens apps, maybe the solution is to move vendor to an `internal` package.
